### PR TITLE
[Bug]: Remove social account.

### PR DESCRIPTION
**Pull request by rhutchison**
The settings controller was moved under settings directory, so I removed the duplicate controller.

There is a bug with removing social accounts due to the undefined "params" on the request.  This is changed to use "query".  While debugging this, I noticed if there are no params passed, the request just times out due to there being no response.  I added some exception handling to throw 401 for unauthenticated user and 400 for no params.

### CODE REVIEW

**Comment by lirantal** [comment]
need someone with more angular experience to review this

File: modules/users/server/controllers/users/users.authentication.server.controller.js [hunk 1]
```diff
@@ -175,33 +175,39 @@ exports.saveOAuthUserProfile = function(req, providerUserProfile, done) {
...
+			req.login(user, function (err) {
+				if (err) {
+					return res.status(400).send(err);
+				} else {
+					return res.json(user);
+				}
+			});
+		}
+	});
+};
```
Comment (ilanbiala) [inline modules/users/server/controllers/users/users.authentication.server.controller.js]: Newline missing

Comment (rhutchison) [inline modules/users/server/controllers/users/users.authentication.server.controller.js]: What's the issue with newline?  Why is it required?

**Comment by ilanbiala** [comment]
@lirantal LGTM. The file that is deleted was just left as a remnant when @amoshaviv split it up into smaller chunks in a folder called settings.

**Comment by lirantal** [comment]
thanks @ilanbiala, merged.

**Pull request merged**